Rubberduck.VBEEditor/ComManagement/ComMessagePumper.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Threading;
+using Rubberduck.VBEditor.ComManagement.VBERuntime;
+
+namespace Rubberduck.VBEditor.ComManagement
+{
+    public static class ComMessagePumper
+    {
+        private static SynchronizationContext UiContext { get; set; }
+        private static IVBERuntime Runtime { get; set; }
+
+        public static void Initialize()
+        {
+            if (UiContext == null)
+            {
+                UiContext = SynchronizationContext.Current;
```

Comment:
that is a difficult assumption right here. Is there any way to guarantee that the current synchronization context is the UI Thread?

Reply:
ATM, it uses the same assumption as `UiDispatcher` does, having its `Initialize` called in both `ParserState.cs` and `App.cs`, which works well enough.

Reply:
Well, one way we could help make the code testable for both this and the `UiDispatcher` is to separate the task of creating the UI context out of the classes and instead have those classes call a method to check they are in the correct context.... ?